metalearners/metalearner.py

```diff
@@ -119,15 +121,18 @@ def _combine_propensity_and_nuisance_specs(
 
 def _filter_x_columns(X: Matrix, feature_set: Features) -> Matrix:
     if feature_set is None:
-        X_filtered = X
-    elif len(feature_set) == 0:
-        X_filtered = np.ones((safe_len(X), 1))
-    else:
-        if isinstance(X, pd.DataFrame):
-            X_filtered = X[list(feature_set)]
-        else:
-            X_filtered = X[:, np.array(feature_set)]
-    return X_filtered
+        return X
+    if len(feature_set) == 0:
+        return np.ones((safe_len(X), 1))
+    if nw.dependencies.is_into_dataframe(X):
+        X_nw = nw.from_native(X)  # type: ignore
+        if all(map(lambda x: isinstance(x, int), feature_set)):
+            columns = [X_nw.columns[index] for index in feature_set]  # type: ignore
+            return X_nw.select(nw.col(column) for column in columns).to_native()
```

Comment:
If you aim for `polars>=1` only, you can use [`nw.nth`](https://narwhals-dev.github.io/narwhals/api-reference/narwhals/#narwhals.nth):

```suggestion
        if all(map(lambda x: isinstance(x, int), feature_set)):
            return X_nw.select(nw.nth(feature_set)).to_native()
```

Reply:
As of now, `feature_set` could be a `Collection[str]`. Based on the doc-string of `nw.nth`, I would presume, that this might not always work, right?

Reply:
I don't fully follow the logic then. Line 129 checks if `feature_set` is a list of (all) integers , if that's the case, then my suggestion is to use `nw.nth` - it will be just syntactic sugar at the end, but it's clear that the selection happens by column index

Reply:
You're absolutely right, of course! Apologies about the confusion. I pushed your suggestion. :)

dd9c07e